java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -99,8 +105,31 @@ abstract class AbstractArcHost(
     // TODO: add lifecycle API for ArcHosts shutting down to cancel running coroutines
     private val scope = CoroutineScope(coroutineContext)
 
+    /** Supports asynchronous [ArcHostContext] serializations. */
+    private val channel: Channel<suspend () -> Unit> = Channel(Channel.UNLIMITED)
+    private val updateArcHostContextScope = CoroutineScope(updateArcHostContextCoroutineContext)
```

Comment:
Is it possible to instead pass in a scope instead of creating one here?

Reply:
It makes sense to pass in a scope if no shutdown nor lifecycle binding can be observed in the `AbstractArcHost`.

The `shutdown()` in the `AbstractArcHost` already takes care of scope cancellation for the `scope` created from the passed-in `coroutineContext `, so I just did the same for `updateArcHostContextScope`.

Any problem you foresee creating a scope in the `AbstractArcHost` and cancelling it in `AbstractArcHost.shutdown`?

Reply:
I'm OK leaving it as is for now, I think we have general work to do on improving our scope creation/passing strategy project-wide. 

The `launchIn` of the flow will return a job that you can cancel on shutdown, so there's no need to save a scope.

Reply:
I think you should leave as-is for now, since it's similar to other patterns we have. 

But, I think this might be in the spirit of Kotlin structured concurrency:
```
AbstractArcHost(
  coroutineScope: CoroutineScope
  arcSerializationDispatcher: CoroutineDispatcher
  ...
) {
  ...
  init {
  ...
   channel.consumeAsFlow().onEach { it() }.launchIn(coroutineScope + arcSerializationDispatcher)
  }
```

This way, there's still just a single scope for structure, but we can use the right dispatcher for the channel/flow/etc.

Reply:
LGTM to central mgmt on scopes, pass in one scope and optionally one or multiple dispatchers.